### PR TITLE
Add Charset. First draft

### LINKED ISSUE #14: HTTP model for Character sets
HTTP uses Internet media types defined in RFC 2046. Some media types such as `text/plain` or `text/html` can optionally declare the character set.

Relevant links:
- [RFC 7231](https://tools.ietf.org/html/rfc7231#section-3.1.1.1)
- [RFC 2046](https://tools.ietf.org/html/rfc2046#section-4.1)

Create a representation for Character sets in `zio.http.Charset`.

### PR DESCRIPTION
**Pull request by saraiva132**
Fixes #14  

In need of some feedback regarding what to call the constructor and the underlying charset type, JCharset seems a bit meh. I mostly based the implementation on java.nio.charset. 

The method apply is a placeholder to what should be a proper name but I could not come up with anything decent.

@ioleo

### CODE REVIEW

**Comment by env0der** [comment]
What is the motivation for putting `java.nio.charset.Charset` inside the case class vs putting a string value?

File: core/src/main/scala/zio/http/Charset.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+final case class Charset private (value: JCharset) extends AnyVal {
+  override def toString: String = value.displayName
+}
+
+object Charset {
+
+  def apply(charSet: JCharset): Charset = Charset(charSet)
+
+  val `ISO-8859-1` = Charset(StandardCharsets.ISO_8859_1)
```
Comment (env0der) [inline core/src/main/scala/zio/http/Charset.scala]: Considering https://github.com/zio/zio-http/pull/12#discussion_r342596937 I propose to use friendly names without backticks.

**Comment by saraiva132** [comment]
Mostly because NIO Charset already implements charset respecting related RFC's. Hence it is valuable to make use of it and wrap it into a ZIO specific data type since internally we will have all the functionality that NIO Charset offers. 

The values presented are basically instances for all standard charsets that are guaranteed to be supported by every implementation of the java platform.

Also note that I was inspired by HTTP4s.

File: core/src/main/scala/zio/http/Charset.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.http
+
+import java.nio.charset.{ StandardCharsets, Charset => JCharset }
+
+final case class Charset private (value: JCharset) extends AnyVal {
```
Comment (ioleo) [inline core/src/main/scala/zio/http/Charset.scala]: Let's make the value's type `String`, so if anyone prefers RFC names can use the constructor.

File: core/src/main/scala/zio/http/Charset.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  override def toString: String = value.displayName
+}
+
+object Charset {
+
+  def apply(charSet: JCharset): Charset = Charset(charSet)
```
Comment (ioleo) [inline core/src/main/scala/zio/http/Charset.scala]: We should map java's Charset to String representation.

File: core/src/main/scala/zio/http/Charset.scala (hunk 1 above)
Comment (ioleo) [reply core/src/main/scala/zio/http/Charset.scala]: Agreed, we should be consistent across the library. Friendly names in the companion and constructor for those that prefer RFC names.

**Comment by saraiva132** [comment]
@ioleo What do you think? Added two smart constructors and changed names to be more friendly

**Comment by saraiva132** [comment]
Hahaha Okay. This naming thing is not my cup of tea. Initially J looked more intuitive but then today I assumed NIO made people relate to java NIO data types..

**Pull request merged**